apps/src/gamelab/blocks.js

```diff
@@ -68,6 +68,34 @@ const customInputTypes = {
       return block.getTitleValue(arg.name);
     },
   },
+  spritePicker: {
+    addInput(block, input) {
+      Blockly.Variables.registerGetter(Blockly.BlockValueType.SPRITE, block.name);
```

Comment:
`registerGetter` marks a block as the one that you want to use in the corresponding autopopulated toolbox category, so it should just be called with `sprite_variables_get`, not every block that has a sprite variable dropdown.

Reply:
Hmm, I'm not sure I fully understand the different options.  I removed this line, but I have the following bug.

1. Drag out a block that uses the `spritePicker` custom input.  Rename the variable:
<img src="https://user-images.githubusercontent.com/413693/39899403-84125f10-5470-11e8-9148-179261884930.png" width="320">
<img src="https://user-images.githubusercontent.com/413693/39899417-992c65da-5470-11e8-81c4-a4d0fd4e6d05.png" width="250">

2. Both variables show up in the "Sprite" toolbox category:
<img src="https://user-images.githubusercontent.com/413693/39899362-53c7505e-5470-11e8-9a59-90e66d774a0b.png" width="130">

But only one of the variable names shows up in the `spritePicker` dropdown: 😕
<img src="https://user-images.githubusercontent.com/413693/39899488-0e2b2466-5471-11e8-8195-f69e5397130d.png" width="320">

Reply:
The sprite category shows you all the sprite variables you've used, along with one additional getter+setter that has a unique name (the normal variable category does this too).

It probably makes less sense to do this for sprite variables. https://github.com/code-dot-org/blockly/pull/105 provides a way to turn that off.

Reply:
Ah, got it.  So this is working as designed 👍

Reply:
I also get the same behavior if I choose "New variable..." in step (1) instead of "Rename variable..." — expected?

Reply:
Yep. There should be one extra pair of variables blocks even before you rename/create a new variable.